backend/cfg/cfg_to_linear.ml

```diff
@@ -345,16 +345,23 @@ let run cfg_with_layout =
   let layout = Array.of_list (CL.layout cfg_with_layout) in
   let len = Array.length layout in
   let next = ref Linear_utils.labelled_insn_end in
+  let tailrec_label = ref None in
   for i = len - 1 downto 0 do
     let label = layout.(i) in
     if not (Label.Tbl.mem cfg.blocks label) then
       Misc.fatal_errorf "Unknown block labelled %d\n" label;
     let block = Label.Tbl.find cfg.blocks label in
     assert (Label.equal label block.start);
     let body =
-      let terminator =
+      let terminator, terminator_tailrec_label =
         linearize_terminator cfg block.terminator ~next:!next
       in
+      (match !tailrec_label, terminator_tailrec_label with
+       | (Some _ | None), None -> ()
+       | None, Some _ -> tailrec_label := terminator_tailrec_label
+       | Some old_trl, Some new_trl ->
+         assert (Label.equal old_trl new_trl);
+         tailrec_label := terminator_tailrec_label);
```

Comment:
If `tailrec_label` and `terminator_tailrec_label` are already equal, why do we need to update `tailrec_label`?